docs/decisions/20240115-Error-handling-report-problem-agent.md

```diff
@@ -0,0 +1,166 @@
+| Verification(Present proof) | https://github.com/decentralized-identity/waci-didcomm/blob/main/present_proof/present-proof-v3.md |
+
+  Custom Behavior table
+  This table defines the expected behavior of the Agent in different scenarios not covered by the specifications.
+
+ | Agent            | Behaviour                      | Comments            |
+ |------------------|--------------------------------|---------------------|
+ | Scenario G1	     | Send a problem report          | e.p.msg.unsupported |
+ | Scenario G2      | Send a problem report          | e.p.msg.unsupported | 
+ | Scenario G3      | Send a problem report          | e.p.error           | 
+ | Scenario G4      | (Sync ?) Send a problem report | e.p.trust.crypto    | 
+ | Scenario G5      | (Sync ?) Send a problem report | e.p.did             | 
+ | Scenario G6      | (Sync ?) Send a problem report | e.p.did.malformed   | 
+ | Scenario G7      | Send a problem report          | e.p.msg.<PIURI>     | 
+
+### Scenarios Description
+
+#### General Scenarios:
+
+- **G1** - Receive a message for an unsupported protocol
+ - G1 - Send a problem report `e.p.msg.unsupported`
+
+- **G2** - Receive a message for an unsupported version of the protocol.
+ - G2 - Send a problem report `e.p.msg.unsupported` and say what version(s) its supported
+
+- **G3** - When an internal error or any unexpected error happens.
+ - G3 - Send a problem report "e.p.error"
+
+- **G4** - If the message is tampered or got any crypto errors when decoding.
+ - G4 -  (sync!) Send a problem report "e.p.trust.crypto"
+
+- **G5** - If the DID method is not supported (`did.peer` in this case)
+ - G5 - (sync?) e.p.did
+
+- **G6** - If the DID method is malformed.
+ - G6 - (sync?) e.p.did.malformed
+
+- **G7** - When a parsing error from the decrypted message.
+ - G7 - Send a problem report `e.p.msg.<PIURI>` if the plaintext message is malformed or if parsing into a specific protocol's data model fails.
+
+##  Connection Flow Scenarios
+https://github.com/hyperledger/aries-rfcs/tree/main/features/0160-connection-protocol              
+
+| Agent             | Behaviour                                           | Action                       |
+|-------------------|-----------------------------------------------------|------------------------------|
+| Scenario C1	      | Send a problem report (Invitation expired)          | e.p.msg.invitation-expired   |
+| Scenario C2       | Send a problem report (Invitation parsing decoding) | e.p.msg.malformed-invitation |
+| Scenario C3       | Send a problem report (DB connection issues)        | e.p.me.res.storage           |
+| Scenario C4       | Send a problem report (After max retries)           | e.p.req.max-retries-exceeded |
+| Scenario C5 (G3)  | Send a problem report Any other error               | e.p.error                    |
+
+- **C1** - OOB Invitation has expired
```

Comment:
These scenario definitions don't match the format of the previous general scenarios - could we have one way of showing these that makes it easier for someone new to the project to reason about them?